packages/cubejs-schema-compiler/adapter/BaseQuery.js

```diff
@@ -1334,7 +1358,9 @@ class BaseQuery {
       sqlUtils: {
         convertTz: this.convertTz.bind(this)
       }
-    }, R.map(
+    },
+    this.safeEvaluateSymbolContext(),
```

Comment:
This one is too broad. Usually it contains just internal of SQL evaluation. It makes sense to use only specific `startDate` and `endDate` properties here.

Reply:
Do you think this will require to write a separate method to handle `startDate` and `endDate`?

Reply:
@bentesha I think something like following will be sufficient:
```
return Object.assign({
      filterParams: this.filtersProxy(),
      sqlUtils: {
        convertTz: this.convertTz.bind(this)
      },
      startDate: this.safeEvaluateSymbolContext().startDate,
      endDate: this.safeEvaluateSymbolContext().endDate
    })
```

Reply:
@paveltiunov why not modify function signature and add a parameter for passing extra context variables? Having to modify function implementation to support new context variables does not seem sustainable in the long term. The function should be flexible enough to accommodate new context variables without modifying its implementation.

```javascript
return Object.assign({
      filterParams: this.filtersProxy(),
      sqlUtils: {
        convertTz: this.convertTz.bind(this)
      }
    }, extraContext )
```

Reply:
@bentesha Yep. Sounds good. We can go this route.

Reply:
@paveltiunov Will make changes and update later today